roles/kubernetes-apps/network_plugin/multus/tasks/main.yml

```diff
@@ -9,9 +9,12 @@
     state: "latest"
   delegate_to: "{{ groups['kube_control_plane'][0] }}"
   run_once: true
-  with_items: "{{ multus_manifest_1.results }} + {{ multus_nodes_list|map('extract', hostvars, 'multus_manifest_2')|list|json_query('[].results') }}"
+  with_items: "{{ multus_manifest_1.results + multus_nodes_list|map('extract', hostvars, 'multus_manifest_2.results')|list }}"
   loop_control:
-    label: "{{ item.item.name }}"
+    label: "{{ item.item.name | default('None') }}"
```

Comment:
Hmmm it's not clear to me why some would not have a name TBH could you elaborate a bit how this could happen?

Reply:
> Hmmm it's not clear to me why some would not have a name/item TBH could you elaborate a bit how this could happen?

well it appeared to me that if the roles/network_plugin/multus/tasks/main.yml is not included ( kube_network_plugin_multus = false ) then vars multus_manifest_1 and multus_manifest_2 are not registered.

also to me it was very confusing the use of item.item. In the provided tasks, there doesn't appear to be any nested loops, so the use of item.item might be incorrect.

Reply:
Ah yes ok I see you are fixing the case when multus is not enabled I should read more carefully x). I don't think you need this label change because `item.item.name` because it's also used above line 4 so supposedly it works (probably?). 

It is indeed weird though that this gets executed I will try to dig why this is even happening...

Reply:
This just doesn't make any sense, are you positive that you didn't enable somewhere `kube_network_plugin_multus` somewhere? Could you like try to dump `kube_network_plugin_multus` in a task somewhere in here?

Something like that:
```
- ansible.builtin.debug:
    var: kube_network_plugin_multus
```

Reply:
Even if its skipped its still evaluated.

I can do the rest you asked tomorrow,  meanwhile i can post complete dump of the error
'''
{{ groups['kube_control_plane'][0] }}] (item=[{'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-crd', 'file': 'multus-crd.yml', 'type': 'customresourcedefinition'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-serviceaccount', 'file': 'multus-serviceaccount.yml', 'type': 'serviceaccount'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-clusterrole', 'file': 'multus-clusterrole.yml', 'type': 'clusterrole'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-clusterrolebinding', 'file': 'multus-clusterrolebinding.yml', 'type': 'clusterrolebinding'}, 'ansible_loop_var': 'item'}] + [[{'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-daemonset-containerd', 'file': 'multus-daemonset-containerd.yml', 'type': 'daemonset', 'engine': 'containerd'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-daemonset-docker', 'file': 'multus-daemonset-docker.yml', 'type': 'daemonset', 'engine': 'docker'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-daemonset-crio', 'file': 'multus-daemonset-crio.yml', 'type': 'daemonset', 'engine': 'crio'}, 'ansible_loop_var': 'item'}]]) => {"ansible_loop_var": "item", "changed": false, "item": "[{'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-crd', 'file': 'multus-crd.yml', 'type': 'customresourcedefinition'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-serviceaccount', 'file': 'multus-serviceaccount.yml', 'type': 'serviceaccount'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-clusterrole', 'file': 'multus-clusterrole.yml', 'type': 'clusterrole'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-clusterrolebinding', 'file': 'multus-clusterrolebinding.yml', 'type': 'clusterrolebinding'}, 'ansible_loop_var': 'item'}] + [[{'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-daemonset-containerd', 'file': 'multus-daemonset-containerd.yml', 'type': 'daemonset', 'engine': 'containerd'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-daemonset-docker', 'file': 'multus-daemonset-docker.yml', 'type': 'daemonset', 'engine': 'docker'}, 'ansible_loop_var': 'item'}, {'changed': False, 'skipped': True, 'skip_reason': 'Conditional result was False', 'item': {'name': 'multus-daemonset-crio', 'file': 'multus-daemonset-crio.yml', 'type': 'daemonset', 'engine': 'crio'}, 'ansible_loop_var': 'item'}]]", "msg": "Failed to template loop_control.label: 'ansible.utils.unsafe_proxy.AnsibleUnsafeText object' has no attribute 'item'. 'ansible.utils.unsafe_proxy.AnsibleUnsafeText object' has no attribute 'item'", "skip_reason": "Conditional result was False"}
'''

From my understanding if kube_network_plugin_multus is not defined or is false, then the tasks within this role should indeed be skipped because the when condition is not met. But note that the when condition is evaluated for each task individually, not for the entire role. So, if there is an error in the tasks within this role, it can still cause the playbook to fail even if the tasks should be skipped.

Reply:
What is also bugging me is that we do have a bunch of CI where multus is disabled so I am a bit confused on why you get that error... Btw you are using either the docker image of kubespray or a venv with the requirements.txt right ? Because we use a pretty old version of Ansible and this task break in more recent version (your with_items fixes that btw)...

Reply:
Check packet_ubuntu20-calico-aio does fail.
Also in other pr for example: https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/-/jobs/4310512382#L3829

Reply:
Oh yeah that make sense, so I advise you to always use a venv with the requirements.txt we provide.

Ansible 6 breaked this `'[1] + {{ [2] }}'` (see https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_6.html#playbook), so IMO what you did with with_items previously looked pretty good, I think it's worth it and it will help people running unsupported version and also help our upgrade of ansible's version of kubespray that should happen relatively soon!